### DISCUSSION TITLE
road to alpha release

### DISCUSSION THREAD

**Discussion by JarbasAl** [OpenVoiceOS/ovos-buildroot#62] [General]
### Open bugs to fix
Below a (short) list of known or otherwise discovered bugs that we would like to fix before we can tag a 0.1.0 version release of the OpenVoiceOS buildroot image release.

- [x] ~Volume not always restored at boot / set to zero.~ Fixed in latest images. 
- [x]  ~TTS not always hearable / outputting. (Looks like this is fixed)~
- [x] ~Mycroft reports ready twice by voice~ Fixed by: Removed accidental included old enclosure skill
- [ ] ...

### Open ToDO's / Additions
Small things still todo or to include before release.

- [x] Add voice feedback to the whole new wifi/network setup. UPDATE: In principle done, however the selections will be converted to dynamic lists.
- [ ] Update TTS pre-loaded cache to reflect the new updated skills.
- [ ] ...

### Basic functionality
Other then the automated function test within the python code / packages it would be best to test the image on a device for basic functionality. Below a list of minimal functionality check for testing the release in real life.

#### Date and Time
- [x] Hey Mycroft, what time is it?
       (expected response: Listening state, current time spoken and shown on screen. Returning to homescreen after idle timeout)
- [x] Hey Mycroft, what is the date?
       (expected response: Listening state, current date spoken and shown on screen. Returning to homescreen after idle timeout)
- [ ] ....

## Hardware tested
Below a list of hardware that has been tested and confirmed working afetr all above bug are fixed and we are ready for release.

- [x] Mycroft Dev-Kit: RPI4, Sj-201 dev HAT board, Internal speakers, Waveshare 5" touchscreen, USB3 disk.
- [ ] ...

**Comment by j1nx** [comment]
### TTS not always hearable / outputting.

```
exception calling callback for <Future at 0x7fa08fe6b0 state=finished raised RuntimeError>
Traceback (most recent call last):
  File "/usr/lib/python3.10/concurrent/futures/_base.py", line 330, in _invoke_callbacks
    callback(self)
  File "/usr/lib/python3.10/site-packages/pyee/_executor.py", line 56, in _callback
    self.emit('error', exc)
  File "/usr/lib/python3.10/site-packages/pyee/_base.py", line 116, in emit
    self._emit_handle_potential_error(event, args[0] if args else None)
  File "/usr/lib/python3.10/site-packages/pyee/_base.py", line 86, in _emit_handle_potential_error
    raise error
  File "/usr/lib/python3.10/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/usr/lib/python3.10/site-packages/mycroft/audio/speech.py", line 62, in handle_speak
    mute_and_speak(utterance, ident, listen)
  File "/usr/lib/python3.10/site-packages/mycroft/audio/speech.py", line 88, in mute_and_speak
    tts.init(bus)
  File "/usr/lib/python3.10/site-packages/ovos_plugin_manager/templates/tts.py", line 521, in init
    TTS.playback.start()
  File "/usr/lib/python3.10/threading.py", line 923, in start
    raise RuntimeError("threads can only be started once")
RuntimeError: threads can only be started once
```

**Comment by j1nx** [comment]
In the latest rolling dev release, it looks like the volume behaves itself when rebooting. However at fresh reboot, the volume is set to 50% but directly sliding the top menu down shows 0% at the slider.
